qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1520A.py

```diff
@@ -840,13 +841,15 @@ def __init__(self, name: str, instrument: B1520A, **kwargs):
         self._fudge: float = 1.5 # fudge factor for setting timeout
 
     def get_raw(self) -> Tuple[Tuple[float, ...], Tuple[float, ...]]:
-        model = self.instrument.impedance_model()
-        if model != constants.IMP.MeasurementMode.Cp_D:
-            raise Exception('Run sweep only supports Cp_D impedance model')
-
         if not self.instrument.setup_fnc_already_run:
             raise Exception('Sweep setup has not yet been run successfully')
 
+        model = self.instrument.impedance_model()
+        names, labels, units = get_name_label_unit_of_impedance_model(model)
```

Comment:
setting name/label/unit in get is too late - users need the correct name/label/unit when they register the parameter. I'd suggest for this multiparameter to have a method along the lines of "update_name_label_unit_of_impedance_model" which will be called once "impedance_mode" value is updated (and the same method can be called on the __init__ of the multiparameter).

This is also needs to be noted in the example notebook - that impedance model has to be set BEFORE registering this parameter.

Reply:
Please have a look on the current implementation.